provider/pkg/provider/agent_pool_test.go

```diff
@@ -24,20 +30,75 @@ func (c *AgentPoolClientMock) CreateAgentPool(ctx context.Context, name, orgName
 func (c *AgentPoolClientMock) UpdateAgentPool(ctx context.Context, agentPoolId, name, orgName, description string) error {
 	return nil
 }
-func (c *AgentPoolClientMock) DeleteAgentPool(ctx context.Context, agentPoolId, orgName string) error {
-	return nil
+func (c *AgentPoolClientMock) DeleteAgentPool(ctx context.Context, agentPoolId, orgName string, forceDestroy bool) error {
+	return c.deleteAgentPoolFunc()
 }
 
-func buildAgentPoolClientMock(getAgentPoolFunc getAgentPoolFunc) *AgentPoolClientMock {
+func buildAgentPoolClientMock(getAgentPoolFunc getAgentPoolFunc, deleteAgentPoolFunc deleteAgentPoolFunc) *AgentPoolClientMock {
 	return &AgentPoolClientMock{
 		getAgentPoolFunc,
+		deleteAgentPoolFunc,
 	}
 }
 
 func TestAgentPool(t *testing.T) {
+	t.Run("Delete without force delete", func(t *testing.T) {
+		errMsg := "Bad Request: 1 stacks have been configured to use deployment pool: \"beep-boop\". " +
+			"Please change the stack deployment configuration before deleting this pool."
+		mockedClient := buildAgentPoolClientMock(
+			nil,
+			func() error {
+				return apitype.ErrorResponse{
+					Code:    400,
+					Message: errMsg,
+				}
+			},
+		)
+
+		provider := PulumiServiceAgentPoolResource{
+			client: mockedClient,
+		}
+
+		req := pulumirpc.DeleteRequest{
+			Id:  "org/abc/beep-boop",
+			Urn: "urn:beep-boop",
+		}
+
+		resp, err := provider.Delete(&req)
```

Comment:
isn't this just testing that the mock returns what you told it to return? 😞
I don't think this is useful

Reply:
Yeah that's fair. Anything that's not a full integration test has been somewhat tricky to test in the provider. Let me spend some time figuring this out.

Reply:
it seems like most of the other tests in the package are the same way 🤷‍♀️ 
if you figure out something I'd love to know for #436 !

Reply:
Yeah I think I'm going to merge this just to unblock Scala but this is a good question that we do need an answer to beyond E2E tests